### PR TITLE
chore: add ensure-npm-valid-dependencies.sh to sanity-checks make target

### PR DESCRIPTION
**Pull request by jviotti**
We forgot to do so on the PR that introduces this script.

This commit also moves all the CI code to separate scripts.

See: https://github.com/resin-io/etcher/pull/1371
Signed-off-by: Juan Cruz Viotti <jviotti@openmailbox.org>

### CODE REVIEW

**Comment by lurch** [comment]
Works on Travis but fails on Appveyor - weird...

**Comment by jviotti** [comment]
Hm, must be the npm version running on Appveyor. I'll make sure they are synced.

File: scripts/ci/travis-install.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+  ./scripts/build/docker/run-command.sh \
+    -r "$TARGET_ARCH" \
+    -s "$(pwd)" \
+    -c "make info && make electron-develop"
+else
+  ./scripts/build/check-dependency.sh pip
+  ./scripts/build/check-dependency.sh brew
+  ./scripts/build/check-dependency.sh make
```
Comment (lurch) [inline scripts/ci/travis-install.sh]: Could check for `npm` too?

File: scripts/ci/appveyor-test.bat [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+:: WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+:: See the License for the specific language governing permissions and
+:: limitations under the License.
+
+IF "%APPVEYOR_REPO_BRANCH%"=="" (
+  ECHO This script is only meant to run in Appveyor CI 1>&2
+  EXIT /B 1
+)
+
+make sanity-checks
```
Comment (lurch) [inline scripts/ci/appveyor-test.bat]: I guess you'll need to add MinGW to the %PATH% before `make` will work?

File: scripts/ci/travis-install.sh (hunk 1 above)
Comment (jviotti) [inline scripts/ci/travis-install.sh]: Done!

File: scripts/ci/appveyor-test.bat (hunk 2 above)
Comment (jviotti) [inline scripts/ci/appveyor-test.bat]: I moved the PATH declarations to `appveyor.yml` so they get re-used everywhere.

**Comment by jviotti** [comment]
> A shame there's so much redundancy (repeated info) between all the scripts, but given the platform-specific and CI-service-specific nature to all this, I guess there's not much that can be done about that.

Yeah, I want to at least try moving everything to bash scripts we can call with MinGW. What do you think? I might do it in another PR soon :)

**Comment by jviotti** [comment]
Looks like we have to prefix any external command we call in a `.bat` file with `call`, otherwise the script exits successfully, but the commands don't actually run :/

**Comment by jviotti** [comment]
All tests finally green :D

File: scripts/ci/appveyor-test.bat (hunk 2 above)
Comment (lurch) [inline scripts/ci/appveyor-test.bat]: Haha, _back_ to `appveyor.yml` ;-)

**Comment by lurch** [comment]
> Yeah, I want to at least try moving everything to bash scripts we can call with MinGW. What do you think?

You could, but there'll still be differences between the setup needed on Appveyor and Travis (and/or differences between OSes), and I was thinking also about the duplication of the "make" commands in the Travis scripts for the Docker and non-Docker versions.

**Pull request merged**